### DISCUSSION TITLE
Consequences of "naïvely" changing fonts

### DISCUSSION THREAD

**Discussion by rosulek** [KaTeX/KaTeX#3585] [Q&A] [answered]
I understand that Katex does not officially support alternative fonts, because Katex needs to know certain metrics about each character. On the other hand, it is very simple to just change the fonts listed in the CSS. I have tried this and the results look reasonable to me. Perhaps I am not typesetting anything terribly complicated. My question is: 

> What specifically should I expect to be "wrong", when simply changing fonts in CSS and making no other changes to Katex?

An additional question is: what would be required to change the font in Katex "the right way", assuming I am willing to put in some effort? I have noticed the [katex-fonts](https://github.com/KaTeX/katex-fonts) repository but it appears to be deprecated. Would this still be the right approach for the latest Katex?

**Comment by edemaine** [comment, marked as answer]
The main issue with different fonts, other than checking that all the Unicode symbols line up (there are often some discrepancies around some symbols), is in the metrics.  See `src/fontMetricsData.js` and the associated build code in `src/metrics`. If these are incorrect for your font, then you'll find vertical alignments to be off.  Try `\frac` and `\begin{matrix}` and so on and the layout will probably look bad.

The font building code is in `src/fonts` in this repository (it moved from `katex-fonts`). You shouldn't need to touch that if you already have a font.

**Comment by ronkok** [comment]
A couple more possible bad consequences are tied to KaTeX's close coupling with its font.

1. `\oiint` and `\oiiint` are composed of an SVG oval superimposed on a KaTeX glyph. A different glyph may be in a position that does not align with the oval.
2.  Tall parentheses and brackets are built out of multiple glyphs and we had complaints that gaps occasionally appeared. To fill the gaps, we've used another small SVG and it is also a risk for misalignment if a different set of glyphs is used.
